contracts/utils/DynamicBuffer.sol

```diff
@@ -0,0 +1,108 @@
+    /// @notice Allocates container space for the DynamicBuffer
+    /// @param capacity The intended max amount of bytes in the buffer
+    /// @return container The memory location of the container
+    /// @return buffer The memory location of the buffer
+    /// @dev Allocates `capacity + 0x60` bytes of space
+    ///      The buffer array starts at the first container data position,
+    ///      (i.e. `buffer = container + 0x20`)
+    function allocate(uint256 capacity)
+        internal
+        pure
+        returns (bytes memory container, bytes memory buffer)
+    {
+        assembly {
+            // Get next-free memory address
+            container := mload(0x40)
+
+            // Allocate memory by setting a new next-free address
+            {
+                // Add 2 x 32 bytes in size for the two length fields
+                // Add 32 bytes safety space for 32B chunked copy
+                let size := add(capacity, 0x60)
+                let newNextFree := add(container, size)
+                mstore(0x40, newNextFree)
+            }
+
+            // Set the correct container length
+            {
+                let length := add(capacity, 0x40)
+                mstore(container, length)
+            }
+
+            // The buffer starts at idx 1 in the container (0 is length)
+            buffer := add(container, 0x20)
+
+            // Init content with length 0
+            mstore(buffer, 0)
+        }
+
+        return (container, buffer);
+    }
+
+    /// @notice Appends data_ to buffer_, and update buffer_ length
+    /// @param buffer_ the buffer to append the data to
+    /// @param data_ the data to append
+    /// @dev Does not perform out-of-bound checks (container capacity)
+    ///      for efficiency.
+    function appendBytes(bytes memory buffer_, bytes memory data_)
+        internal
+        pure
+    {
+        assembly {
+            let length := mload(data_)
+            for {
+                let data := add(data_, 32)
+                let dataEnd := add(data, length)
+                let buf := add(buffer_, add(mload(buffer_), 32))
+            } lt(data, dataEnd) {
+                data := add(data, 32)
+                buf := add(buf, 32)
+            } {
+                // Copy 32B chunks from data to buffer.
+                // This may read over data array boundaries and copy invalid
+                // bytes, which doesn't matter in the end since we will
+                // later set the correct buffer length.
+                mstore(buf, mload(data))
+            }
+
+            // Update buffer length
+            mstore(buffer_, add(mload(buffer_), length))
+        }
+    }
+
+    /// @notice Appends data_ to buffer_, and update buffer_ length
+    /// @param buffer_ the buffer to append the data to
+    /// @param data_ the data to append
+    /// @dev Performs out-of-bound checks and calls `appendBytes`.
+    function appendBytesSafe(bytes memory buffer_, bytes memory data_)
```

Comment:
In the above example, there is absolutely no reason to take the underscored version. This was simply the result of copying the function signature from `appendBufferUnchecked`.

In the other case we had colliding names (now removed) between a local variable and function arguments. Here, the style guide suggests the use of a single trailing underscore https://docs.soliditylang.org/en/v0.8.9/style-guide.html#avoiding-naming-collisions .

On second thought, however, I think it should be preferred to change the variable that occurs later (i.e. the local variable instead of the function argument). Since the official style guide is ambiguous on which on to change, I'd like to suggest this as an internal standard.

Reply:
Good to know re the style guide, thank you!